### PR TITLE
Add `New chat` handler to assistant UI chat 

### PR DESCRIPTION
**Pull request by rSnapkoOpenOps**
Fixes OPS-2205.

## Testing Checklist

Check all that apply:

- [X] I tested the feature thoroughly, including edge cases

- [X] I verified all affected areas still work as expected

- [X] Automated tests were added/updated if necessary

- [X] Changes are backwards compatible with any existing data, otherwise a migration script is provided

https://www.loom.com/share/a98dbe4ec2894d61b986e8fae8dff5a7?sid=1c31121a-05cc-4c85-834b-79f14c1ebd42

### CODE REVIEW

File: packages/react-ui/src/app/features/ai/assistant-ui/assistant-ui-chat.tsx [hunk 1]
```diff
@@ -67,6 +74,31 @@ const AssistantUiChat = ({
...
+    } catch (error) {
+      toast({
+        title: t('There was an error creating the new chat, please try again'),
+        duration: 3000,
+      });
+      console.error(
+        `There was an error deleting existing chat and creating a new one: ${error}`,
+      );
+    }
+  }, [queryClient]);
```
Comment (rmn-snpk) [inline packages/react-ui/src/app/features/ai/assistant-ui/assistant-ui-chat.tsx]: Copied from AI assistant and removed 'delete' function for old chat

File: packages/react-ui/src/app/features/ai/assistant-ui/assistant-ui-chat.tsx [hunk 2]
```diff
@@ -67,18 +74,53 @@ const AssistantUiChat = ({
...
+    } catch (error) {
+      toast({
+        title: t('There was an error creating the new chat, please try again'),
+        duration: 3000,
+      });
+      console.error(
+        `There was an error deleting existing chat and creating a new one: ${error}`,
+      );
+    }
+  }, [queryClient, runtime.thread]);
```
[the remark this answers is not in the document]
Comment (rmn-snpk) [reply packages/react-ui/src/app/features/ai/assistant-ui/assistant-ui-chat.tsx:104]: Not relevant

File: packages/react-ui/src/app/features/ai/assistant-ui/assistant-ui-chat.tsx [hunk 3]
```diff
@@ -67,18 +74,53 @@ const AssistantUiChat = ({
...
 
+  const runtime = useChatRuntime(runtimeConfig);
+  const [hasMessages, setHasMessages] = useState(!!openChatResponse?.messages);
+
+  const queryClient = useQueryClient();
+
+  const createNewChat = useCallback(async () => {
+    const oldChatId = chatId.current;
+
+    chatId.current = null;
```
Comment (MarceloRGonc) [inline packages/react-ui/src/app/features/ai/assistant-ui/assistant-ui-chat.tsx:85]: Is it acceptable to do this even if an exception occurs later?

Comment (rmn-snpk) [reply packages/react-ui/src/app/features/ai/assistant-ui/assistant-ui-chat.tsx:85]: Yep, it's fine. This will reset the chat id. If error occurs the user will get a new chat after reload
